app/src/main/java/com/example/dungeontest/EditorScreen.kt

```diff
@@ -0,0 +1,105 @@
+package com.example.dungeontest
+
+import android.graphics.BitmapFactory
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material.icons.filled.Menu
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.DrawerState
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.FabPosition
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavController
+import com.example.dungeontest.model.EditorViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun EditorScreen(drawerState: DrawerState, scope: CoroutineScope, navController: NavController) {
+    val snackbarHostState = remember { SnackbarHostState() }
+
+    val keyboardController = LocalSoftwareKeyboardController.current
+    val focusManager = LocalFocusManager.current
+    val viewModel = viewModel<EditorViewModel>()
+    val byteArray = viewModel.byteArrayOfImage.collectAsState().value
+    val bitmapImage = byteArray?.let {
```

Comment:
```suggestion
    val bitmapImage = byteArray.let {
```
IDE says `?` unnecessary here

Reply:
more generally, since byteArray isn't nullable, why not eliminate the let-call approach altogether?

Reply:
However, if this code is all going to disappear (after we start using AsyncImage and the Coil library), please disregard these comments

Reply:
was experimenting with the  return value being potentially null. Please ignore for now as it's not final and will likely change